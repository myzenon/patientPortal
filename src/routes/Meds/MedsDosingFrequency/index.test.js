import React from 'react'
import { MedsDosingFrequency } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {},
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        }
    }
    const wrapper = render(
        <MedsDosingFrequency { ...props } />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test.skip('clicks NEXT button with input the questions | api fhirCmrIdPut | go to Meds 2 Verify Drug', (done) => {
    const mockQuestions = "Can I have this pill with meal ?"
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {}
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                medication_000080833: {
                    taken: "n",
                    reasonNotTaken: {},
                    question: mockQuestions,
                    reviewCompleted: true
                }
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                medication_000080833: {
                    taken: "n",
                    reasonNotTaken: {},
                    question: mockQuestions,
                    reviewCompleted: true
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/meds-verify-drug/000293159')
            done()
        }
    }
    const wrapper = mount(
        <MedsDosingFrequency  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('textarea[type="text"]').simulate('change', {
            target: {
                name: 'question',
                value: mockQuestions
            }
        })
        wrapper.find('button').simulate('click')
    }, 0)
})

test.skip('clicks NEXT button without input the questions | api fhirCmrIdPut | go to Meds 2 Verify Drug', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {}
            }
        },
        match: {
            params: {
                medId: '000080833'
            }
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                medication_000080833: {
                    taken: "n",
                    reasonNotTaken: {},
                    reviewCompleted: true
                }
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                medication_000080833: {
                    taken: "n",
                    reasonNotTaken: {},
                    reviewCompleted: true
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/meds-verify-drug/000293159')
            done()
        }
    }
    const wrapper = mount(
        <MedsDosingFrequency  { ...props } />
    )
    setTimeout(() => {
        wrapper.find('button').simulate('click')
    }, 0)
})

test.skip('clicks NEXT button goes to OtherMeds | all meds are completed', (done) => {
    const props = {
        cmr: {
            ...mockObj.cmr,
            medication_000080833: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: true
            },
            medication_000293159: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: true
            },
            medication_000493990: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: true
            },
            medication_000747126: {
                taken: "n",
                reasonNotTaken: {},
                reviewCompleted: true
            },
            medication_665820312: {
                taken: "n",
                reasonNotTaken: {},
            },
        },
        match: {
            params: {
                medId: '665820312'
            }
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirCmrIdPut')
            expect(get).toEqual({
                id: mockObj.cmr.id
            })
            expect(post).toEqual({
                medication_665820312: {
                    taken: "n",
                    reasonNotTaken: {},
                    reviewCompleted: true
                }
            })
            return Promise.resolve()
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual({
                ...mockObj.cmr,
                medication_000080833: {
                    taken: "n",
                    reasonNotTaken: {},
                    reviewCompleted: true
                },
                medication_000293159: {
                    taken: "n",
                    reasonNotTaken: {},
                    reviewCompleted: true
                },
                medication_000493990: {
                    taken: "n",
                    reasonNotTaken: {},
                    reviewCompleted: true
                },
                medication_000747126: {
                    taken: "n",
                    reasonNotTaken: {},
                    reviewCompleted: true
                },
                medication_665820312: {
                    taken: "n",
                    reasonNotTaken: {},
                    reviewCompleted: true
                }
            })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/prepare/Chapter3')
            done()
        }
    }
    const wrapper = mount(
        <MedsDosingFrequency { ...props } />
    )
    setTimeout(() => {
        wrapper.find('button').simulate('click')
    }, 0)
})
