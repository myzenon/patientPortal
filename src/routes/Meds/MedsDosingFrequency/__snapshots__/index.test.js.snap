// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <br />
    <span
      class="heading1"
    >
      EFFEXOR XR 75MG
    </span>
    <div
      class="questions"
    >
      How often do you take this medicine?
    </div>
    <div
      class="contentText"
    >
      <span
        class="bulletPoints"
      >
        <li>
          Click "Every Day", if you take this medication daily
        </li>
        <p />
        <li>
          Click "As Needed", if you take this medication only when you need it
        </li>
        <p />
        <li>
          Click "Other", if you take this medication any other way
        </li>
      </span>
    </div>
    <div
      class="contentButton"
    >
      <table>
        <tbody>
          <tr>
            <td>
              <button
                class="button btn btn-secondary"
                id="everyDayButton"
                type="button"
              >
                Every Day
              </button>
            </td>
            <td>
                 
            </td>
            <td>
              <button
                class="button btn btn-secondary"
                id="asNeededButton"
                type="button"
              >
                As Needed
              </button>
            </td>
            <td>
                 
            </td>
            <td>
              <button
                class="button btn btn-secondary"
                id="otherButton"
                type="button"
              >
                Other
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <br />
     
    <br />
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepCompleted"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="stepActive"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumberActive"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
